### PR TITLE
Adds a Nudist Quirk

### PR DESCRIPTION
**Pull request by MilkForever**
# About The Pull Request
Adds a new neutral quirk called "Nudist", this quirk gives you an invisible jumpsuit in your backpack that you can wear, as the name suggests, its a jumpsuit that's becomes invisible when worn, doesn't actually make you invisible.

It still behaves like a normal jumpsuit, so you still have pockets, suit sensors, a belt and an ID card slot.

Special thanks to dwasint for helping me to get the jumpsuit to spawn in backpacks.

## Why It's Good For The Game
I've seen several people asking for a Nudist Quirk so here we are, it doesn't impact gameplay and it's more for customization than anything else.

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

image
</details>

## Changelog

:cl:
add: Added the Nudist Quirk! This is a neutral quirk that gives you an invisible jumpsuit, for those who don't like clothing.
/:cl:

### CODE REVIEW

File: code/datums/traits/neutral.dm [hunk 1]
```diff
@@ -97,3 +97,20 @@
 
 /datum/quirk/monochromatic/remove()
 	quirk_holder.remove_client_colour(/datum/client_colour/monochrome)
+//Monkestation change: Nudist Quirk by MilkForever, special thanks to dwasint
+/datum/quirk/nudist
```
Comment (Zonespace27) [inline code/datums/traits/neutral.dm:101]: this can be modularized

File: code/datums/traits/neutral.dm [hunk 2]
```diff
@@ -97,3 +97,20 @@
...
+	name = "Nudist"
+	desc = "You managed to get a rare invisible jumpsuit, it stills works as a normal jumpsuit."
+	value = 0
+	gain_text = "<span class='notice'>You have an invisible jumpsuit in your backpack</span>"
+	lose_text = "<span class='danger'>You... don't have an invisible jumpsuit?</span>"
+
+/datum/quirk/nudist/on_spawn()
+    var/mob/living/carbon/human/H = quirk_holder
```
Comment (Zonespace27) [inline code/datums/traits/neutral.dm:109]: one letter vars bad

File: code/modules/clothing/under/miscellaneous.dm [hunk 3]
```diff
@@ -129,6 +129,16 @@
 	icon_state = "plasmarefill"
 	icon = 'icons/obj/device.dmi'
 
+//Monkestation change: Nudist quirk, invisible jumpsuit by MilkForever
+/obj/item/clothing/under/invisible
```
Comment (Zonespace27) [inline code/modules/clothing/under/miscellaneous.dm:133]: this can be modularized

File: code/modules/clothing/under/miscellaneous.dm [hunk 4]
```diff
@@ -129,6 +129,16 @@
...
+	name = "invisible jumpsuit"
+	desc = "Some sort of jumpsuit that dissapears when worn, but still behaves as a normal jumpsuit."
```
Comment (Zonespace27) [inline code/modules/clothing/under/miscellaneous.dm:135]:
```suggestion
	desc = "Some sort of jumpsuit that disappears when worn, but still behaves as a normal jumpsuit."
```

**Review by Zonespace27 — CHANGES_REQUESTED** [review]
as much as i dislike this quirk in its entirety for more reasons than one (not that i can do anything about it), code review

**Comment by MilkForever** [comment]
Alright, let me see if i can uh, modularize this so its in the monkestation folder, i think that's ideal.

**Comment by MilkForever** [comment]
Wait, dang. im gonna draft this for now.

**Comment by MilkForever** [comment]
Okay the checks passed now, but im still not sure if this is ready.

**Review by nednaZ — CHANGES_REQUESTED** [review]

File: monkestation/code/datums/traits/neutral.dm [hunk 5]
```diff
@@ -40,3 +40,17 @@
...
+	value = 0
+
+/datum/quirk/nudist/on_spawn()
+    var/mob/living/carbon/human/person = quirk_holder
+    var/obj/item/clothing/under/invisible/clothing = new(get_turf(person))
+    var/list/slots = list (
+        "backpack" = ITEM_SLOT_BACKPACK,
+        "hands" = ITEM_SLOT_HANDS,
+    )
+    person.equip_in_one_of_slots(clothing , slots , qdel_on_fail = TRUE)
```
[the remark this answers is not in the document]
Comment (MilkForever) [reply monkestation/code/datums/traits/neutral.dm]: Re-added the gain_text now.

**Review by nednaZ — APPROVED** [review]
I'm willing to give this a try, only because Ook has requested this quirk.

**Pull request merged**